core/src/apps/management/recovery_device/homescreen.py

```diff
@@ -174,15 +204,42 @@
 
 async def _request_share_next_screen(ctx: wire.Context, mnemonic_type: int) -> None:
     if mnemonic_type == mnemonic.TYPE_SLIP39:
-        remaining = storage.recovery.get_remaining()
+        remaining = storage.recovery.fetch_slip39_remaining_shares()
+        group_count = storage.recovery.get_slip39_group_count()
         if not remaining:
             # 'remaining' should be stored at this point
             raise RuntimeError
-        if remaining == 1:
-            text = "1 more share"
+
+        if group_count > 1:
+            content = layout.RecoveryHomescreen(
+                "More shares needed", "for this recovery"
+            )
+            await layout.homescreen_dialog(ctx, content, "Enter share")
         else:
-            text = "%d more shares" % remaining
-        content = layout.RecoveryHomescreen(text, "needed to enter")
-        await layout.homescreen_dialog(ctx, content, "Enter share")
+            if remaining == 1:
```

Comment:
seems this should be `if len(remaining) == 1`

Reply:
remaining stores a List of ints storing remaining value fore each group. so it should actually be remaining[0]. Good catch.